app/repositories/players.py

```diff
@@ -169,72 +214,71 @@ async def fetch_many(
         params["limit"] = page_size
         params["offset"] = (page - 1) * page_size
 
-    recs = await app.state.services.database.fetch_all(query, params)
-    return [dict(rec) for rec in recs]
+    players = await app.state.services.database.fetch_all(query, params)
+    return cast(list[Player], players) if players is not None else None
 
 
 async def update(
     id: int,
-    name: str | None = None,
-    email: str | None = None,
-    priv: int | None = None,
-    country: str | None = None,
-    silence_end: int | None = None,
-    donor_end: int | None = None,
-    creation_time: int | None = None,
-    latest_activity: int | None = None,
-    clan_id: int | None = None,
-    clan_priv: int | None = None,
-    preferred_mode: int | None = None,
-    play_style: int | None = None,
-    custom_badge_name: str | None = None,
-    custom_badge_icon: str | None = None,
-    userpage_content: str | None = None,
-    api_key: str | None = None,
+    name: str | _UnsetSentinel = UNSET,
+    email: str | _UnsetSentinel = UNSET,
+    priv: int | _UnsetSentinel = UNSET,
+    country: str | _UnsetSentinel = UNSET,
+    silence_end: int | _UnsetSentinel = UNSET,
+    donor_end: int | _UnsetSentinel = UNSET,
+    creation_time: _UnsetSentinel | _UnsetSentinel = UNSET,
+    latest_activity: int | _UnsetSentinel = UNSET,
+    clan_id: int | _UnsetSentinel = UNSET,
+    clan_priv: int | _UnsetSentinel = UNSET,
+    preferred_mode: int | _UnsetSentinel = UNSET,
+    play_style: int | _UnsetSentinel = UNSET,
+    custom_badge_name: str | _UnsetSentinel = UNSET,
+    custom_badge_icon: str | _UnsetSentinel = UNSET,
```

Comment:
this can actually be null
```suggestion
    custom_badge_icon: str | None | _UnsetSentinel = UNSET,
```

https://github.com/osuAkatsuki/bancho.py/blob/c426e50fa87376d21a988b9461a302845148332f/migrations/base.sql#L307